qiskit_nature/problems/second_quantization/molecular/fermionic_op_builder.py

```diff
@@ -0,0 +1,105 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+""" Fermionic operator builder. """
+
+import itertools
+
+import numpy as np
+
+from qiskit_nature.drivers.qmolecule import QMolecule
+from qiskit_nature.operators import FermionicOp
+
+
+def build_fermionic_op(q_molecule: QMolecule) -> FermionicOp:
+    """
+    Builds a fermionic operator based on a QMolecule object.
+
+    Args:
+        q_molecule (QMolecule): QMolecule instance with 1- and/or 2-body integrals.
+
+    Returns:
+        FermionicOp: FermionicOp built from a QMolecule object.
+    """
+
+    one_body_ints = q_molecule.one_body_integrals
+    two_body_ints = q_molecule.two_body_integrals
+
+    return build_ferm_op_from_ints(one_body_ints, two_body_ints)
+
+
+def build_ferm_op_from_ints(one_body_integrals: np.ndarray,
+                            two_body_integrals: np.ndarray = None) -> FermionicOp:
+    """
+    Builds a fermionic operator based on 1- and/or 2-body integrals.
+    This method requires the integrals stored in the '*chemist*' notation
+             h2(i,j,k,l) --> adag_i adag_k a_l a_j
+    and the integral values are used for the coefficients of the second-quantized
+    Hamiltonian that is built. The integrals input here should be in block spin
+    format and also have indexes reordered as follows 'ijkl->ljik'
```

Comment:
The required to be stored in chemist notation has been confusing in the past. FermionicOperator had similar language. Certainly if we are building out the block spin format, from integrals that were originally in chemist notation, then to have them compatible with the way they were used as coeffs for input for the operator buildout the indexes needed the reordering. But the input here does not care how the integrals were originally stored as long as they are in the right order when presented here.

Reply:
Do you suggest any specific change in this case? Here I mostly migrated what was already done elsewhere in the project.

Reply:
I think this text that it states the it **requires** the integrals stored in chemist notation. Its more like if we have a source that is two body integrals stored in chemist notation then it needs the ijkl->ljik  reordering as well as altering to be in this block spin format. Also since we treated these more as coeffs for the build out of the 2nd quantized operator there is a 0.5 factor applied to the coeff. For me its the **requires** where it is - what we require is the final block spin with the ordering we expect etc. The source we do not require chemist notation ordered integrals, but of course you can start from there, which is in fact what we have in our normal path.